spec/services/change_course_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+
+RSpec.describe ChangeCourse do
+  include CourseOptionHelpers
+
+  let(:application_choice) { create(:application_choice, status: :awaiting_provider_decision) }
+  let(:provider_user) do
+    create(
+      :provider_user,
+      :with_make_decisions,
+      providers: [application_choice.current_course_option.provider],
+    )
+  end
+  let(:course_option) { course_option_for_provider(provider: application_choice.current_course_option.provider, course: application_choice.current_course_option.course) }
+
+  let(:change_course) do
+    described_class.new(
+      actor: provider_user,
+      application_choice: application_choice,
+      course_option: course_option,
+    )
+  end
+
+  describe '#save!' do
+    describe 'if the actor is not authorised to perform this action' do
+      let(:provider_user) do
+        create(:provider_user,
+               providers: [application_choice.current_course_option.provider])
+      end
+
+      it 'throws an exception' do
+        expect {
+          change_course.save!
+        }.to raise_error(
+          ProviderAuthorisation::NotAuthorisedError,
+          'You do not have the required user level permissions to make decisions on applications for this provider.',
+        )
+      end
+    end
+
+    describe 'if the provided details are correct' do
+      let(:application_choice) { create(:application_choice, status: :awaiting_provider_decision) }
+      let(:provider_user) do
+        create(:provider_user,
+               :with_make_decisions,
+               providers: [application_choice.course_option.provider])
+      end
+
+      it 'then it calls the services' do
+        allow(application_choice).to receive(:update_course_option_and_associated_fields!)
+
+        change_course.save!
+
+        expect(application_choice).to have_received(:update_course_option_and_associated_fields!)
```

Comment:
I think we need to be specific with the fields being updated here, rather than checking just the method. Can do in a subsequent PR

Reply:
ok!